### PR TITLE
Cosmin s/tema6

### PR DESCRIPTION
**Pull request by Cosmin0589**

### CODE REVIEW

**Review by sorinfoca — APPROVED** [review]

File: src/main/java/cosminsechelmain/driver/BrowserManager.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package cosminsechelmain.driver;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class BrowserManager {
```
Comment (95teo) [inline src/main/java/cosminsechelmain/driver/BrowserManager.java:6]: No need to create another package with this class. You already have the one from homework 5. Remove this one and use that one instead. Thanks!

File: src/main/java/cosminsechelmain/utils/FileUtils.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package cosminsechelmain.utils;
```
Comment (95teo) [inline src/main/java/cosminsechelmain/utils/FileUtils.java:1]: Add this class in the existing cosminshechel package inside utils. No need to create cosminsechelmain package. Thanks!

File: src/test/java/cosminsechel1/tests/tema6/DemoQaWindow.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
+package cosminsechel1.tests.tema6;
```
Comment (95teo) [inline src/test/java/cosminsechel1/tests/tema6/DemoQaWindow.java:1]: Put tema6 package inside cosminsechel.tests package. No nee to create cosminsechel1 package. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/test/java/cosminsechel/tests/tema6/DemoQaWindow.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import org.openqa.selenium.interactions.Actions;
+import cosminsechel.utils.FileUtils;
+import cosminsechel.driver.BrowserManager;
+
+import java.io.File;
+import java.util.Set;
+
+public class DemoQaWindow {
+    static ChromeDriver driver = null;
+    private static File screenshot;
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/tema6/DemoQaWindow.java:16]: You can remove this line as it's never used. Thanks!

File: src/test/java/cosminsechel/tests/tema6/DemoQaWindow.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    public static void main(String[] args) {
+        navigateToDemoQaWindowPage();
+        try {
+            clickNewTabButton();
+            interactWithElementsOnParentTab();
+        } catch (NoSuchElementException e) {
+            FileUtils.takeScreenshot(driver, "window");
+            System.out.println("No alert present");
+        } finally {
+            closeBrowser();
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/tema6/DemoQaWindow.java:27]: Use method from BrowserManager to close the driver. Thanks!

File: src/test/java/cosminsechel/tests/tema6/DemoQaWindow.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        WebElement droppableButton = driver.findElement(By.id("droppable"));
+        action.dragAndDrop(draggableButton, droppableButton).build().perform();
+        WebElement droppedText = driver.findElement(By.id("dropped"));
+        System.out.println("Dropped text: " + droppedText.getText());
+        WebElement sliderButton = driver.findElement(By.id("slider"));
+        action.dragAndDropBy(sliderButton, 50, 0).build().perform();
+        WebElement sliderText = driver.findElement(By.id("sliderValue"));
+        System.out.println("Slider text: " + sliderText.getText());
+    }
+    public static void closeBrowser() {
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/tema6/DemoQaWindow.java:89]: After you added the method from BrowserManager to close the driver delete this method from here. Thanks!

File: src/test/java/cosminsechel/tests/tema6/HerokuAlerts.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    public static void findNonexistentElement() {
+        try {
+            WebElement nonExistentElement = driver.findElement(By.id("nonexistent-element"));
+        } catch (NoSuchElementException e) {
+            FileUtils.takeScreenshot(driver, "nonexistent_element");
+            System.out.println("Could not find the element with id 'nonexistent-element'");
+        }
+    }
+
+    public static void closeBrowser() {
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/tema6/HerokuAlerts.java:102]: After you added the method from BrowserManager to close the driver delete this method from here. Thanks!

File: src/test/java/cosminsechel/tests/tema6/HerokuForms.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        WebElement submitButton = driver.findElement(By.xpath("//input[@type='submit']"));
+        Actions actions = new Actions(driver);
+        actions.moveToElement(submitButton).build().perform();
+    }
+    public static void submitForm() {
+        driver.findElement(By.xpath("//input[@type='submit']")).click();
+        System.out.println("Form submitted");
+    }
+
+    public static void closeBrowser() {
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/tema6/HerokuForms.java:70]: After you added the method from BrowserManager to close the driver delete this method from here. Thanks!

File: src/test/java/cosminsechel/tests/tema6/HerokuAlerts.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+            handleDifferentTypeOfAlert();
+            findNonexistentElement();
+        } catch (NoAlertPresentException e) {
+            FileUtils.takeScreenshot(driver, "alert");
+            System.out.println("No alert present");
+        } catch (NoSuchElementException e) {
+            FileUtils.takeScreenshot(driver, "nonexistent_element");
+            System.out.println("Could not find the element with id 'nonexistent-element'");
+        } finally {
+            closeBrowser();
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/tema6/HerokuAlerts.java:32]: Use method from BrowserManager to close the driver. Thanks!

File: src/test/java/cosminsechel/tests/tema6/HerokuForms.java [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        navigateToHerokuHomePage();
+        try {
+            fillOutForm();
+            scrollToSubmitButton();
+            submitForm();
+        } catch (NoSuchElementException e) {
+            FileUtils.takeScreenshot(driver, "alert");
+            System.out.println("No alert present");
+        } finally {
+            closeBrowser();
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/tema6/HerokuForms.java:28]: Use method from BrowserManager to close the driver. Thanks!

File: src/test/java/cosminsechel/tests/tema6/HerokuForms.java [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.interactions.Actions;
+
+import java.io.File;
+
+public class HerokuForms {
+
+    static ChromeDriver driver = null;
+    private static File screenshot;
```
Comment (95teo) [inline src/test/java/cosminsechel/tests/tema6/HerokuForms.java:16]: You can remove this line as it's never used. Thanks!

**Review by 95teo — APPROVED** [review]

**Pull request merged**